payload/game/ui/Section.cc

```diff
@@ -496,6 +496,57 @@ void Section::logDebuggingInfo(bool verbose) {
             m_pages[i]->logControlsDebug();
         }
     }
+    for (int i = static_cast<int>(PageId::EXT_MININCLUSIVE__);
+            i < static_cast<int>(PageId::EXT_MAXEXCLUSIVE__); ++i) {
+        logPageInfo(m_pages[i]);
+        if (m_pages[i]) {
+            m_pages[i]->logControlsDebug();
+        }
+    }
+}
+
+extern "C" void Page_unfocusMaybe(Page *);
+
+void Section::deinit() {
+    // Since we zero out the arrays, we can trust REPLACED(deinit) does not double-free.
+    for (u32 i = m_activePageCount; i > 0; --i) {
+        auto &page = m_activePages[i - 1];
+        Page_unfocusMaybe(page);
+        page->deactivate();
+        page = nullptr;
+    }
```

Comment:
```suggestion
    cullActivePages(0);
```
We can use the existing base game helper function at 806232e8.

The behavior here is also incorrect, as we do not set the page to a nullptr in this loop.

Reply:
I don't think the behavior is incorrect in this context. We can use the helper, though.

Reply:
If the page is a nullptr, `onDeinit` is never called in the next loop.

Reply:
This is irrelevant because `m_activePageCount` will be zero in all cases here, so the values of `m_activePages` will never be inspected. With respect to the other arrays, we rely on the zeroing to avoid a double-free bug

Reply:
Oh, sorry, I misunderstood that setting the page to a nullptr would affect `m_pages` and not `m_activePages`. Yes, you're right.